core/src/main/java/org/elasticsearch/index/query/QueryBuilders.java

```diff
@@ -40,13 +40,20 @@
 public abstract class QueryBuilders {
 
     /**
-     * A query that match on all documents.
+     * A query that matches on all documents.
      */
     public static MatchAllQueryBuilder matchAllQuery() {
         return new MatchAllQueryBuilder();
     }
 
     /**
+     * A query that matches no documents.
+     */
+    public static MatchNoneQueryBuilder matchNoneQuery() {
```

Comment:
Yes we can remove that one. We just make the usage of this builder undocumented.